### PR TITLE
console elements

### PR DESCRIPTION
**Pull request by victorbalan**
- Load challenge using challenge id from cookies
  when the app loads
- Change login check
- Add logic for loading the correct element using
  the web interface from a challenge
- Add unit test console
- Add simple console
- Add engine_url in utils
- Add new dependency: iron-icon

### CODE REVIEW

File: app/scripts/util.js [hunk 1]
```diff
@@ -8,3 +8,5 @@ function getCookie(cname) {
   }
   return "";
 }
+
+var engine_url = "http://localhost:8081"
```
Comment (lorenzleutgeb) [inline app/scripts/util.js:12]: i think we should not hardcode this. As the URL of engine is not the same as the origin of this file, and we decided not to reverse proxy through to the engine, we should deliver the base URL of engine via some API. What do you think?

File: app/scripts/app.js [hunk 2]
```diff
@@ -38,4 +50,15 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
     }
   };
 
+  app.persistChallenge = function(challenge){
+    app.challenge = challenge;
+    document.cookie = "challengeId="+challenge.id;
```
Comment (lorenzleutgeb) [inline app/scripts/app.js:55]: Isn't this a very good scenario to use the LocalStorage API?

File: app/scripts/util.js (hunk 1 above)
Comment (victorbalan) [inline app/scripts/util.js:12]: i don't really see the point..we`ll always know the engine url .

File: app/scripts/app.js (hunk 2 above)
Comment (victorbalan) [inline app/scripts/app.js:55]: we will add an expiring date to the challengeId so the user cannot access that challenge after some time so in my opinion the cookies are the best storage option.

**Comment by victorbalan** [comment]
Need refactoring to match the new api.

**Pull request closed without merging**